mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -631,6 +654,32 @@ class RandomForestSuite extends SparkFunSuite with MLlibTestSparkContext {
     val expected = Map(0 -> 1.0 / 3.0, 2 -> 2.0 / 3.0)
     assert(mapToVec(map.toMap) ~== mapToVec(expected) relTol 0.01)
   }
+
+  test("[SPARK-3159] tree model redundancy - binary classification") {
+    redundacyTest(2, 20)
+  }
+
+  test("[SPARK-3159] tree model redundancy - multiclass classification") {
+    redundacyTest(4, 20)
+  }
+
+  private def redundacyTest(numClasses: Int, numPoints: Int) = {
+    val strategy = new OldStrategy(algo = Classification, impurity = Gini, maxDepth = 4,
+      numClasses = numClasses, maxBins = 32)
+
+    val rnd = new Random(100)
+
+    val rdd = RandomForestSuite.generateRedundantPoints(sc,
+      numClasses = numClasses, numPoints = numPoints, rnd)
+
+    val dt = RandomForest.run(rdd, strategy, numTrees = 1, featureSubsetStrategy = "auto",
+      seed = 42, instr = None).head
+
+    val isRedundant = RandomForestSuite.isRedundant(dt)
```

Comment:
`isRedundant` seems ... redundant here. Just `assert(!RandomForestSuite.isRedundant(dt))`. Similar comment for other similar constructs around here.

Reply:
You are right, I have removed isRedundant and rnd.